### ISSUE TITLE
[BUG] The new implementation of SBX seems wrong

### ISSUE THREAD

**Issue by MLopez-Ibanez** [esa/pagmo2#438]
**Describe the bug**
Pull request #436 changed SBX. I believe the change is wrong and it doesn't match the SBX in https://www.iitk.ac.in/kangal/codes/nsga2/nsga2-gnuplot-v1.1.6.tar.gz

**Comment by darioizzo** [comment]
And what makes you believe that?

**Comment by darioizzo** [comment]
https://github.com/esa/pagmo2/blob/master/src/utils/genetic_operators.cpp#L101 here a few lines were lost in the refactoring, changing the distribution. The following is the corrected version:
```
                rand01 = drng(random_engine);
                beta = 1. + (2. * (y1 - yl) / (y2 - y1));
                alpha = 2. - std::pow(beta, -(eta_c + 1.));
                if (rand01 < (1. / alpha)) {
                    betaq = std::pow((rand01 * alpha), (1. / (eta_c + 1.)));
                } else {
                    betaq = std::pow((1. / (2. - rand01 * alpha)), (1. / (eta_c + 1.)));
                }
                c1 = 0.5 * ((y1 + y2) - betaq * (y2 - y1));
                beta = 1. + (2. * (yu - y2) / (y2 - y1));
                alpha = 2. - std::pow(beta, -(eta_c + 1.));
                if (rand01 < (1. / alpha)) {
                    betaq = std::pow((rand01 * alpha), (1. / (eta_c + 1.)));
                } else {
                    betaq = std::pow((1. / (2. - rand01 * alpha)), (1. / (eta_c + 1.)));
                }
                c2 = 0.5 * ((y1 + y2) + betaq * (y2 - y1));
```

**Comment by MLopez-Ibanez** [comment]
See pr #439

**Comment by MLopez-Ibanez** [comment]
Also, the changes to SBX and polynomial mutation operators are not described in the Changelog for 2.16: https://esa.github.io/pagmo2/changelog.html

Those changes seem newsworthy to me. People may get different results in their algorithms after updating.

**Comment by bluescarni** [comment]
@MLopez-Ibanez perhaps you can add the entry in the changelog in #439?

**Comment by MLopez-Ibanez** [comment]
Done

**Comment by bluescarni** [comment]
Fixed by #439.
